godot-ffi/src/global.rs

```diff
@@ -23,9 +26,10 @@ use std::sync::{Mutex, MutexGuard, TryLockError};
 ///
 /// There are two `const` methods for construction: [`new()`](Self::new) and [`default()`](Self::default).
 /// For access, you should primarily use [`lock()`](Self::lock). There is also [`try_lock()`](Self::try_lock) for special cases.
-pub struct Global<T> {
+pub struct Global<T, F = fn() -> T> {
     // When needed, this could be changed to use RwLock and separate read/write guards.
-    value: Mutex<InitState<T>>,
+    value: Mutex<OnceCell<T>>,
+    init_fn: F,
 }
```

Comment:
The `F` parameter is not necessary; `new()` in its current form can only be invoked with a function pointer. I'd keep this a pure refactoring without changing the public API, we can discuss new features separately if needed.